rts/motoko-rts/src/memory/ic/linear_memory.rs

```diff
@@ -6,18 +6,18 @@ use crate::types::*;
 /// Amount of garbage collected so far.
 pub(crate) static mut RECLAIMED: Bytes<u64> = Bytes(0);
 
-// Heap pointer
+// Heap pointer (skewed)
 extern "C" {
-    pub(crate) fn setHP(new_hp: u32);
+    pub(crate) fn setHP(new_hp: u32); // usize????
     pub(crate) fn getHP() -> u32;
 }
 
 /// Heap pointer after last GC
 pub(crate) static mut LAST_HP: u32 = 0;
 
 pub(crate) unsafe fn initialize() {
-    setHP(get_aligned_heap_base());
-    LAST_HP = getHP();
+    LAST_HP = get_aligned_heap_base();
```

Comment:
maybe `LAST_HP` can be skewed too?

Reply:
This could make sense for similarity. Maybe the same for the heap base pointer. As for performance, this pointer is accessed only a very few times during the classical GCs...

Reply:
I left it unskewed but changed it to `usize` (with millions of knock-on changes). Hopefully this will help the 64-bit port sometime.